pycqed/instrument_drivers/meta_instrument/heterodyne.py

```diff
@@ -23,75 +23,295 @@ class HeterodyneInstrument(Instrument):
         - Add power settings
         - Add integration time settings
         - Build test suite
-        - Add parameter Heterodyne voltage (that returns a complex value/ 2 values)
+        - Add parameter Heterodyne voltage (that returns a complex value / two
+          values)
         - Add different demodulation settings.
         - Add fading plots that shows the last measured avg transients
           and points in the IQ-plane in the second window
         - Add option to use CBox integration averaging mode and verify
            identical results
-    '''
+    """
     shared_kwargs = ['RF', 'LO', 'AWG']
 
-    def __init__(self, name,  RF, LO, AWG=None, acquisition_instr=None, acquisition_instr_controller=None,
+    def __init__(self, name, RF, LO, AWG=None, acquisition_instr=None,
+                 acquisition_instr_controller=None,
                  single_sideband_demod=False, **kw):
-        logging.info(__name__ + ' : Initializing instrument')
-        Instrument.__init__(self, name, **kw)
 
-        self.LO = LO
         self.RF = RF
-        self.AWG = AWG
-        self.add_parameter('frequency',
-                           label='Heterodyne frequency',
-                           units='Hz',
-                           get_cmd=self.do_get_frequency,
-                           set_cmd=self.do_set_frequency,
-                           vals=vals.Numbers(9e3, 40e9))
-        self.add_parameter('f_RO_mod', parameter_class=ManualParameter,
-                           vals=vals.Numbers(-600e6, 600e6),
-                           label='Intermodulation frequency',
-                           units='Hz', initial_value=10e6)
+
+        self.common_init(name, LO, AWG, acquisition_instr,
+                         single_sideband_demod, **kw)
+
         self.add_parameter('RF_power', label='RF power',
-                           units='dBm',
+                           units='dBm', vals=vals.Numbers(),
                            set_cmd=self.do_set_RF_power,
                            get_cmd=self.do_get_RF_power)
-        self.add_parameter('single_sideband_demod',
-                           label='Single sideband demodulation',
-                           parameter_class=ManualParameter)
-        self.add_parameter('acquisition_instr',
-                           set_cmd=self._do_set_acquisition_instr,
-                           get_cmd=self._do_get_acquisition_instr,
-                           vals=vals.Anything())
         self.add_parameter('acquisition_instr_controller',
-                           set_cmd=self._do_set_acquisition_instr_controller,
-                           get_cmd=self._do_get_acquisition_instr_controller,
+                           set_cmd=self.do_set_acquisition_instr_controller,
```

Comment:
I found the prefixes needlessly long (eg `_do_get`), but I did not think about the public/private aspect. I changed the private functions to `_get` and `_set` now.